### PR TITLE
Change tidb_opt_projection_push_down default value to ON

### PR DESCRIPTION
**Pull request by yibin87**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)

v8.3: Changes the default value of `tidb_opt_projection_push_down` from `OFF` to `ON`.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/17891
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by windtalker — APPROVED** [review]
LGTM

File: system-variables.md [hunk 1]
```diff
@@ -4487,11 +4487,16 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
 - Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
 - Type: Boolean
-- Default value: `OFF`
-- Specifies whether to allow the optimizer to push `Projection` down to the TiKV or TiFlash coprocessor.
+- Default value: `ON`
+- Specifies whether to allow the optimizer to push `Projection` down to the TiKV coprocessor. Currently only three types of `Projection` operators are allowed to be pushed down:
+    - The top-level expressions of the operator are all JSON query class or JSON value attribute class functions, such as `JSON_EXTRACT`, `JSON_DEPTH`.
+    - The top-level expressions of the operator are partly JSON query class or JSON value attribute class functions, and partly direct column reads.
+    - The top-level expressions of the operator are all direct column reads, and the number of columns output is less than the number of columns input.
+- Whether the `Projection` operator is finally pushed down or not depends on the optimizer's comprehensive evaluation of performance.
```
Comment (Oreoxmt) [inline system-variables.md:4498]:
```suggestion
- Default value: `ON`. Before v8.3.0, the default value is `OFF`.
- Specifies whether to allow the optimizer to push `Projection` down to the TiKV coprocessor. When enabled, the optimizer might push the following three types of `Projection` `Projection` operators down to TiKV:
    - The top-level expressions of the operator are all [JSON query functions](/functions-and-operators/json-functions/json-functions-search.md) or [JSON value attribute functions](/functions-and-operators/json-functions/json-functions-return.md). For example: `SELECT JSON_EXTRACT(data, '$.name') FROM users;`.
    - The top-level expressions of the operator include a mix of JSON query functions or JSON value attribute functions, and direct column reads. For example: `SELECT JSON_DEPTH(data), name FROM users;`.
    - The top-level expressions of the operator are all direct column reads, and the number of output columns is less than the number of input columns. For example: `SELECT name FROM users;`.
- The final decision to push down a `Projection` operator depends on the optimizer's comprehensive evaluation of performance.
```

File: system-variables.md [hunk 2]
```diff
@@ -4487,11 +4487,16 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
 
 ### tidb_opt_projection_push_down <span class="version-mark">New in v6.1.0</span>
 
-- Scope: SESSION
+- Scope: SESSION | GLOBAL
```
Comment (Oreoxmt) [inline system-variables.md:4490]:
```suggestion
- Scope: SESSION | GLOBAL
- Persists to cluster: Yes/No
```

**Review by Oreoxmt — APPROVED** [review]

File: system-variables.md [hunk 3]
```diff
@@ -4487,11 +4487,17 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
+- Default value: `ON`. Before v8.3.0, the default value is `OFF`.
+- Specifies whether to allow the optimizer to push `Projection` down to the TiKV coprocessor. When enabled, the optimizer might push the following three types of `Projection` `Projection` operators down to TiKV:
+    - The top-level expressions of the operator are all [JSON query functions](/functions-and-operators/json-functions/json-functions-search.md) or [JSON value attribute functions](/functions-and-operators/json-functions/json-functions-return.md). For example: `SELECT JSON_EXTRACT(data, '$.name') FROM users;`.
+    - The top-level expressions of the operator include a mix of JSON query functions or JSON value attribute functions, and direct column reads. For example: `SELECT JSON_DEPTH(data), name FROM users;`.
+    - The top-level expressions of the operator are all direct column reads, and the number of output columns is less than the number of input columns. For example: `SELECT name FROM users;`.
+- The final decision to push down a `Projection` operator depends on the optimizer's comprehensive evaluation of performance.
```
Comment (Oreoxmt) [inline system-variables.md:4499]:
```suggestion
- The final decision to push down a `Projection` operator depends on the optimizer's comprehensive evaluation of query cost.
```

File: system-variables.md [hunk 4]
```diff
@@ -4487,11 +4487,17 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
+- This variable is disabled by default for TiDB clusters that are upgraded from a version earlier than v8.3.0 to v8.3.0 or later.
```
Comment (Oreoxmt) [inline system-variables.md:4500]:
```suggestion
- For TiDB clusters that are upgraded from a version earlier than v8.3.0 to v8.3.0 or later, this variable remains the setting before the upgrade.
```

File: system-variables.md [hunk 5]
```diff
@@ -4487,11 +4487,17 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
 
-- Scope: SESSION
+- Scope: SESSION | GLOBAL
+- Persists to cluster: Yes
 - Applies to hint [SET_VAR](/optimizer-hints.md#set_varvar_namevar_value): Yes
 - Type: Boolean
-- Default value: `OFF`
-- Specifies whether to allow the optimizer to push `Projection` down to the TiKV or TiFlash coprocessor.
+- Default value: `ON`. Before v8.3.0, the default value is `OFF`.
+- Specifies whether to allow the optimizer to push `Projection` down to the TiKV coprocessor. When enabled, the optimizer might push the following three types of `Projection` `Projection` operators down to TiKV:
```
Comment (qiancai) [inline system-variables.md:4495]:
```suggestion
- Specifies whether to allow the optimizer to push the `Projection` operator down to the TiKV coprocessor. When enabled, the optimizer might push the following three types of `Projection` operators down to TiKV:
```

File: system-variables.md [hunk 6]
```diff
@@ -4487,11 +4487,17 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
+    - The top-level expressions of the operator are all [JSON query functions](/functions-and-operators/json-functions/json-functions-search.md) or [JSON value attribute functions](/functions-and-operators/json-functions/json-functions-return.md). For example: `SELECT JSON_EXTRACT(data, '$.name') FROM users;`.
+    - The top-level expressions of the operator include a mix of JSON query functions or JSON value attribute functions, and direct column reads. For example: `SELECT JSON_DEPTH(data), name FROM users;`.
+    - The top-level expressions of the operator are all direct column reads, and the number of output columns is less than the number of input columns. For example: `SELECT name FROM users;`.
+- The final decision to push down a `Projection` operator depends on the optimizer's comprehensive evaluation of query cost.
```
Comment (qiancai) [inline system-variables.md:4499]:
```suggestion
- The final decision to push down a `Projection` operator also depends on the optimizer's comprehensive evaluation of query cost.
```

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: system-variables.md [hunk 7]
```diff
@@ -4487,11 +4487,17 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
+    - The top-level expressions of the operator are all [JSON query functions](/functions-and-operators/json-functions/json-functions-search.md) or [JSON value attribute functions](/functions-and-operators/json-functions/json-functions-return.md). For example: `SELECT JSON_EXTRACT(data, '$.name') FROM users;`.
+    - The top-level expressions of the operator include a mix of JSON query functions or JSON value attribute functions, and direct column reads. For example: `SELECT JSON_DEPTH(data), name FROM users;`.
+    - The top-level expressions of the operator are all direct column reads, and the number of output columns is less than the number of input columns. For example: `SELECT name FROM users;`.
+- The final decision to push down a `Projection` operator also depends on the optimizer's comprehensive evaluation of query cost.
+- For TiDB clusters that are upgraded from a version earlier than v8.3.0 to v8.3.0 or later, this variable remains the setting before the upgrade.
```
Comment (yibin87) [inline system-variables.md:4500]:
```suggestion
- For TiDB clusters that are upgraded from a version earlier than v8.3.0 to v8.3.0 or later, this variable will be set to `OFF`.
```

File: system-variables.md [hunk 8]
```diff
@@ -4487,11 +4487,17 @@ EXPLAIN FORMAT='brief' SELECT COUNT(1) FROM t WHERE a = 1 AND b IS NOT NULL;
...
+- Specifies whether to allow the optimizer to push the `Projection` operator down to the TiKV coprocessor. When enabled, the optimizer might push the following three types of `Projection` operators down to TiKV:
+    - The top-level expressions of the operator are all [JSON query functions](/functions-and-operators/json-functions/json-functions-search.md) or [JSON value attribute functions](/functions-and-operators/json-functions/json-functions-return.md). For example: `SELECT JSON_EXTRACT(data, '$.name') FROM users;`.
+    - The top-level expressions of the operator include a mix of JSON query functions or JSON value attribute functions, and direct column reads. For example: `SELECT JSON_DEPTH(data), name FROM users;`.
+    - The top-level expressions of the operator are all direct column reads, and the number of output columns is less than the number of input columns. For example: `SELECT name FROM users;`.
+- The final decision to push down a `Projection` operator also depends on the optimizer's comprehensive evaluation of query cost.
+- For TiDB clusters that are upgraded from a version earlier than v8.3.0 to v8.3.0 or later, this variable will be set to `OFF`.
```
Comment (Oreoxmt) [inline system-variables.md:4500]:
```suggestion
- For TiDB clusters that are upgraded from a version earlier than v8.3.0 to v8.3.0 or later, the default value of this variable is `OFF`.
```

**Review by qiancai — APPROVED** [review]

**Pull request merged**